// @ampt/sdk/dist/index.js
import { createRequire as topLevelCreateRequire } from 'module'
const require = topLevelCreateRequire(import.meta.url || __dirname)
import {
  SocketManager
} from "./chunk-ZO5WVBKJ.js";
import {
  core
} from "./chunk-T2SRD37U.js";
import {
  http
} from "./chunk-OMAPTMVS.js";
import {
  params
} from "./chunk-J3IDRJ24.js";
import {
  schedule,
  task
} from "./chunk-LD42NL2W.js";
import "./chunk-SN7XVJE5.js";
import {
  storage
} from "./chunk-JWWLTU5R.js";
import "./chunk-K3AK6KJP.js";
import "./chunk-LUDTQ6LK.js";
import "./chunk-2Y2UCMIB.js";
import "./chunk-4PLIU73S.js";
import "./chunk-DJQ5LOTZ.js";
import "./chunk-U3BU4F3V.js";
import "./chunk-F7P3IZWD.js";
import "./chunk-QBXM46RJ.js";
import "./chunk-WSNLJX2S.js";
import {
  events
} from "./chunk-IQPEN25V.js";
import "./chunk-2E5TW4HQ.js";
import "./chunk-6XEJQOQ7.js";
import "./chunk-MFAT3D7D.js";
import "./chunk-R467GWIW.js";
import "./chunk-DVAUNSLU.js";
import "./chunk-6I7VLTO4.js";
import "./chunk-HWSCZVIB.js";

// index.ts
if (!global.addEventListener) {
  global.addEventListener = () => {
  };
}
var ws = new SocketManager();
export {
  core,
  events,
  http,
  params,
  schedule,
  storage,
  task,
  ws
};
